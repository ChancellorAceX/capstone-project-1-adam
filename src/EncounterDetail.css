section.encountercontainer {
  display:flex;
  flex-direction: column;
  align-items: center;
  width:inherit;
}

.encounterbody {
  margin-top:7.5vh;
  width:97vw;
  height:80vh;
  display:flex;
  flex-direction: column;
  align-items: center;
}

.encountermembers * {
  text-align: center;
}
