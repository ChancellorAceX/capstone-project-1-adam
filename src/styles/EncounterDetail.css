 main.encounterbody{
  position:fixed;
  top:10vh;
  width:calc(100vw + 20px);
  height:82vh;
  background-color: black;
  color:#aaaaaa;
  overflow:auto;
  display:flex;
  flex-direction: column;
  align-items: center;
}

 table.encountermembers {
  margin:0;
  width:100vw;
} 

table.encountermembers tr{
  width:95vw;
}

table.encountermembers th.viewLink,td.viewLink{
  width:calc(12.5vw + 3px);
  padding:0;
  margin:0;
}

table.encountermembers th.name,td.name{
  width:30vw;
  margin:0;
  padding:0;
}

table.encountermembers th.init,td.init{
  width:10vw;
  margin:0;
  padding:0;
}

table.encountermembers th.ac,td.ac {
  width:10vw;
  margin:0;
  padding:0;
}

table.encountermembers th.hp,td.hp {
  width: 15vw;
  margin:0;
  padding:0;
}

table.encountermembers th.buttons,td.buttons {
  width:calc(12.5vw + 3px);
  margin:0;
  padding:0;

}

table.encountermembers a.viewLink {
  width:12.5vw;
  height:5vh;
  border:0;
  padding:0;
}

table.encountermembers td {
  margin-left:1px;
  padding:5px;
  text-align: center;
}

button.editButton,button.viewLink,button.saveEdit {
  border:1px solid #444444;
  width:12.5vw;
  height:5vh;
  font-weight: 600;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444;
}

/*Edit Stylings*/

#editname,#editinit,#editac,#edithp,#editbuttons {
  padding:0;
}

textarea.pcname {
  width:20vw;
  margin:0;
  word-wrap: normal;
}

input.initiative {
  width:8vw;
  margin:0;
}

input.ac {
  width:8vw;
  margin:0;
}

input.hp,input.hpmax {
  width:10vw;
  max-width:60px;
  margin:0;
}

button.saveEdit {
  height: 64.78px;
}