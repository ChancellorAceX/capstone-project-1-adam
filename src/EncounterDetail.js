import React from 'react';
import './EncounterDetail.css';

export default function EncounterDetail(){
  return (
    <section className='encountercontainer'>
      <header className='detailheader'>
        <button className='back'>back button</button>
        <h2>Encounter Title</h2>
        <button className='add to encounter'>add button</button>
      </header>
      <body className='encounterbody'>
        <p>Encounter Description</p>
        <table className='encountermembers'>
          <tr>
            <th>Name</th>
            <th>Initiative</th>
            <th>AC</th>
            <th>HP</th>
          </tr>
          <tr>
            <td>Monster 1</td>
            <td>16</td>
            <td>20</td>
            <td>58/63</td>
          </tr>
          <tr>
            <td>Player 1</td>
            <td>12</td>
            <td>15</td>
            <td>23/28</td>
          </tr>
          <tr>
            <td>Player 2</td>
            <td>10</td>
            <td>22</td>
            <td>19/25</td>
          </tr>
          <tr>
            <td>Monster 2</td>
            <td>15</td>
            <td>13</td>
            <td>30/42</td>
          </tr>
        </table>
      </body>
    </section>
  )
}