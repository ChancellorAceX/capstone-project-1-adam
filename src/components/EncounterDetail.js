import React from 'react';
import { withRouter } from 'react-router';
import '../styles/EncounterDetail.css';
import TokenService from '../services/token-service';
import CharacterEncounterListing from './CharacterEncounterListing';
import Header from './Header';
import config from '../config';

export default withRouter(class EncounterDetail extends React.Component {
  componentDidMount() {
    const eid = this.props.match.params.eid;
    fetch(`${config.API_ENDPOINT}/api/encounter/${eid}`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res => res.json())
      .then(encounterData => this.props.setEncounterData(encounterData))
      .catch(error => console.log({ error }));
  }

  render() {
    return (
      <section className='encountercontainer'>
        <Header 
          pageName='Encounter Detail'
          history={this.props.history}
        />
        <main className='encounterbody'>
          <p>{this.props.encounters.find(e=>e.eid=== +this.props.match.params.eid).encounterdesc}</p>
          <table className='encountermembers'>
            <tbody>
              <tr>
                <th></th>
                <th>Name</th>
                <th>Init</th>
                <th>AC</th>
                <th>HP</th>
              </tr>
              {this.props.encounterData.sort((a,b)=>b.initiative-a.initiative).map((character,i) =>
                <CharacterEncounterListing
                  key={character.pcname+` ${i}`}
                  character={character}
                  updateCharacter={this.props.updateCharacter}
                  setUpdateObject={this.props.setUpdateObject}
                  encounterData={this.props.encounterData}
                  setTargetState={this.props.setTargetState}
                  index={i}
                />
              )}
            </tbody>
          </table>
        </main>
      </section>
    );
  }
});