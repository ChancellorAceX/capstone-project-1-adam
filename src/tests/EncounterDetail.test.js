import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EncounterDetail from '../components/EncounterDetail';

describe('EncounterDetail component', () => {
  const props = {
    "encounters": [
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 6, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 7, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}"
    ],
    "setEncounterData": "ƒ () {}",
    "encounterData": [
      "{ac: 50, hp: 72, hpmax: 100, initiative: 30, level:…}",
      "{ac: 12, hp: 24, hpmax: 24, initiative: 5, level: n…}",
      "{ac: 12, hp: 15, hpmax: 20, initiative: null, level…}",
      "{ac: 18, hp: 32, hpmax: 32, initiative: null, level…}"
    ],
    "setUpdateObject": "ƒ () {}",
    "updateCharacter": "ƒ () {}",
    "setTargetState": "ƒ () {}",
    "history": {
      "length": 4,
      "action": "PUSH",
      "location": "{hash: \"\", key: \"741myf\", pathname: \"/encounter/1\",…}",
      "createHref": "ƒ createHref() {}",
      "push": "ƒ push() {}",
      "replace": "ƒ replace() {}",
      "go": "ƒ go() {}",
      "goBack": "ƒ goBack() {}",
      "goForward": "ƒ goForward() {}",
      "block": "ƒ block() {}",
      "listen": "ƒ listen() {}"
    },
    "loggedInToggle": "ƒ () {}",
    "location": {
      "pathname": "/encounter/1",
      "search": "",
      "hash": "",
      "key": "741myf"
    },
    "match": {
      "path": "/encounter/:eid",
      "url": "/encounter/1",
      "isExact": true,
      "params": "{eid: \"1\"}"
    }
  };

  it('renders the full encounter detail component', () => {
    const wrapper = shallow(<EncounterDetail {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});