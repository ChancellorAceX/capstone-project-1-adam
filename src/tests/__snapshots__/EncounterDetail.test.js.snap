// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncounterDetail component renders the full encounter detail component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
